### PR TITLE
ref: Allows for more flexibility in track creation.

### PR DESCRIPTION
**Pull request by gpolitis**
Provides https://github.com/jitsi/jitsi-videobridge/pull/589

### CODE REVIEW

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java [hunk 1]
```diff
@@ -761,6 +755,20 @@ FrameDesc getLastReceivedFrame()
...
     {
+        if (!baseIsInitialized)
+        {
+            if (ArrayUtils.isNullOrEmpty(dependencyEncodings))
+            {
+                base = this;
+            }
+            else
+            {
+                base = dependencyEncodings[0].getBaseLayer();
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: Just a general concern: signaling a circular dependency will result in an infinite loop here. At some layer between the client and this code (maybe jicofo, maybe jvb) we need to ensure this doesn't happen.

Comment (bgrozev) [reply src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: Looks like the bridge ensure this.

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java [hunk 2]
```diff
@@ -761,6 +755,20 @@ FrameDesc getLastReceivedFrame()
      */
     public RTPEncodingDesc getBaseLayer()
     {
+        if (!baseIsInitialized)
```
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: can checking `base` for null replace the need for this extra state variable?

**Comment by gpolitis** [comment]
https://github.com/jitsi/jitsi-videobridge/pull/589 has been closed.

**Pull request closed without merging**